src/aspnetcore/eng/build.ps1

```diff
@@ -479,6 +484,7 @@ try {
             /p:Configuration=Release `
             /p:Restore=$RunRestore `
             /p:Build=true `
+            /p:DotNetBuildRepo=$ProductBuild `
```

Comment:
This one should now also only be passed in when ProductBuild is explicitly provided. Same for the one in build.sh